modules/spacelift-spaces/main.tf

```diff
@@ -0,0 +1,72 @@
+resource "spacelift_space" "stage" {
+  for_each = toset(local.stages)
+
+  name        = each.key
+  description = "This a child of the `root` space. It contains all the resources common to the `${each.key}` infrastructure.  (Managed by Terraform)."
+
+  # Every account has a root space that serves as the root for the space tree.
+  # Except for the root space, all the other spaces must define their parents
+  parent_space_id  = "root"
+  inherit_entities = true
+}
+
+resource "spacelift_space" "account" {
+  for_each = local.account_info_map
```

Comment:
This assumes a space per account. What if you wanted a space per tenant or wanted a custom set of spaces?